arangod/Pregel/PregelFeature.cpp

```diff
@@ -87,29 +241,38 @@ void PregelFeature::beginShutdown() {
   Instance = nullptr;
 }
 
-void PregelFeature::addConductor(std::unique_ptr<Conductor>&& c, uint64_t executionNumber) {
+void PregelFeature::addConductor(std::unique_ptr<Conductor>&& c,
+                                 uint64_t executionNumber) {
   MUTEX_LOCKER(guard, _mutex);
-  //_executions.
-  _conductors.emplace(executionNumber, std::shared_ptr<Conductor>(c.get()));
+  std::string user = ExecContext::CURRENT ? ExecContext::CURRENT->user() : "";
+  _conductors.emplace(
+      executionNumber,
+      std::make_pair(user, std::shared_ptr<Conductor>(c.get())));
```

Comment:
couldn't you store the user on the conductor ?

Reply:
You could, yes. It's fairly arbitrary which way to do it. This matches the way I implemented it in the other handlers, as there were some structures that I didn't necessarily want to change elsewhere.